lib/daru/io/io.rb

```diff
@@ -202,23 +200,16 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
         ::CSV
-          .read(path, 'rb',opts)
-          .tap { |c| yield c if block_given? }
+          .parse(open(path), opts)
           .by_col.map { |col_name, values| [col_name, values] }.to_h
       end
 
       def from_csv_hash(path, opts)
         csv_as_arrays =
-          ::CSV
-          .open(path, 'rb', opts)
-          .tap { |c| yield c if block_given? }
```

Comment:
@v0dro I am wondering, if anybody relies on this behavior (e.g. an ability to pass block for preprocessing the data)? It is untested and [not really documented](http://www.rubydoc.info/gems/daru/0.1.5/Daru/DataFrame#from_csv-class_method) so probably we could consider just dropping it?..